src/gtimelog/main.py

```diff
@@ -319,6 +321,14 @@ def on_edit_log(self, action, parameter):
         filename = Settings().get_timelog_file()
         self.open_in_editor(filename)
 
+    def on_edit_last_item(self, action, parameter):
+        window = self.get_active_window()
+        text = window.timelog.remove_last_entry()
+        if text is not None:
+            window.task_entry.set_text(text)
+        window.task_entry.grab_focus()
+        window.task_entry.select_region(-1, -1)
```

Comment:
> Should this be an app-global action rather than a window action?

I do not know what's the difference. Does app global mean it gets an entry in the menu?

Reply:
`gocept.gtimelog` had no history browsing. So I did not consider it during the implementation. I think if previous days cannot be edited (= adding entries), deletion should not happen, when the user is not looking at the current day. Removing the last entry from the current day when looking at a different one, I'd consider unexpected.

Reply:
> I do not know what's the difference. Does app global mean it gets an entry in the menu?

No, this is a [purely internal distinction](https://docs.gtk.org/gtk4/actions.html#action-scopes).  Window actions get dispatched to the current window, and seeing as the first thing you do in the global action is find the current window to do things with its widgets, this seems like it would be more natural to move the handler method into the window subclass.

> I think if previous days cannot be edited (= adding entries), deletion should not happen, when the user is not looking at the current day. Removing the last entry from the current day when looking at a different one, I'd consider unexpected.

Yeah, that makes sense.